src/v/kafka/server/handlers/list_partition_reassignments.cc

```diff
@@ -0,0 +1,183 @@
+              replicas.push_back(bshard.node_id);
+          });
+
+        if (reassignments_by_topic.contains(ntp.tp.topic)) {
+            reassignments_by_topic[ntp.tp.topic].partitions.push_back(
+              ongoing_partition_reassignment{
+                .partition_index = ntp.tp.partition, .replicas = replicas});
+        } else {
+            ongoing_topic_reassignment topic_reassignment{.name = ntp.tp.topic};
+            topic_reassignment.partitions.push_back(
+              ongoing_partition_reassignment{
+                .partition_index = ntp.tp.partition, .replicas = replicas});
+            reassignments_by_topic[ntp.tp.topic] = topic_reassignment;
+        }
+    }
+
+    std::vector<ongoing_topic_reassignment> all_active_reassignments;
+    for (auto& [_, reassignments] : reassignments_by_topic) {
+        all_active_reassignments.push_back(reassignments);
+    }
+
+    return all_active_reassignments;
+}
+
+template<>
+ss::future<response_ptr> list_partition_reassignments_handler::handle(
+  request_context ctx, [[maybe_unused]] ss::smp_service_group ssg) {
+    list_partition_reassignments_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    list_partition_reassignments_response resp;
+
+    if (!ctx.authorized(
+          security::acl_operation::describe, security::default_cluster_name)) {
+        resp.data.error_code = error_code::cluster_authorization_failed;
+        resp.data.error_message = ss::sstring{
+          error_code_to_str(error_code::cluster_authorization_failed)};
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    // TODO(@NyaliaLui) - Check if we are the controller
+
+    // If topics is undefined, get all active reassignments
+    if (!request.data.topics.has_value()) {
+        auto all_active_reassignments = all_active_partition_reassignments(
+          ctx.metadata_cache().updates_in_progress());
+        resp.data.topics.reserve(all_active_reassignments.size());
+        std::for_each(
+          all_active_reassignments.begin(),
+          all_active_reassignments.end(),
+          [&resp](const ongoing_topic_reassignment& topic_reassignment) {
+              resp.data.topics.push_back(topic_reassignment);
+          });
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    resp.data.topics.reserve(request.data.topics->size());
+    for (const list_partition_reassignments_topics& tp : *request.data.topics) {
+        ongoing_topic_reassignment topic_reassignment{.name = tp.name};
+
+        for (const model::partition_id& pid : tp.partition_indexes) {
+            model::ntp ntp{model::kafka_namespace, tp.name, pid};
+
+            vlog(klog.debug, "Request to list reassignments: ntp {}", ntp);
+
+            // Current replica set is in the current partition assignment
+            auto current_assignment
+              = ctx.metadata_cache().get_partition_assignment(ntp);
```

Comment:
same here

Reply:
Makes sense. I can get the shard that an `ntp` is on with the shard_table.